### DISCUSSION TITLE
What is the equivalent of `getScripts` in App Router?

### DISCUSSION THREAD

**Discussion by webcolor-in** [vercel/next.js#60123] [App Router]
I want to customize auto-injected `script` tag in `head` tag eg: add custom `data-*` attribute to each tag, set `nonce` etc. In Page Router we have `getScripts()` but what is the equivalent in App Router?

Example: 
```
class CustomHead extends Head {
  getScripts(files) {
    const originalScripts = super.getScripts(files);
    return originalScripts.map((script) => {
      return React.cloneElement(script, { crossOrigin: 'anonymous' });
    });
  }
}

**Comment by hexionaut** [comment]
Hi.

If you want to inject custom things in `head` using app router the better word around currently is to do same as i do to inject prev and next page meta tags.

Just use the metadata api and the `icons` attribute.

```
nextAndPrevPages.push({
  rel: 'prev',
  url: prevPage === 1 ? getAppURL(page.story.slug) : getAppURL(`${page.story.slug}/page/${prevPage}`),
});

icons: {
  ...sharedIconsMetadata,
  other: getNextAndPrevPages(),
},
```

**Reply by webcolor-in** [reply]
If you can re-write my code that is compatible with App Router?
